person.rb

```diff
@@ -0,0 +1,47 @@
+class Person
+  def initialize(id, age, name: 'Unknown', parent_permission: true)
+    @id = id
+    @id = id
+    @age = age
+    @parent_permission = parent_permission
+    @name = name
+  end
+
+  # Setter method for id
+  def add_id(id)
+    @id = id
+  end
+
+  # Setter method for name
+  def add_name=(name)
+    @name = name
+  end
+
+  # Setter method for age
+  def add_age=(age)
+    @age = age
+  end
+
+  # Getter method for id
+  attr_reader :id
+
+  # Getter method for name
+  attr_reader :name
+
+  # Getter method for age
+  attr_reader :age
```

Comment:
- [ ] @MuhammadUsmanAslam great work implementing the changes requested however, adding `attr_reader :name` and ` attr_reader :age` negates the implementation to have setters for age and name in you above method, I would suggest a simpler solution of getting rid of the methods and using the inbuild ruby methods for setters and readers, also some of your comments above the methods are misleading instead of  Getter method for age/name it should be  Setter method for age/name
```suggestion
  # *:name and :age instance variables are the only ones that should be able to be modified that's why they are defined as attr_accessors (here both Getters and Setters are defined)
  attr_accessor :name, :age 
  # *:id should only be able to be retrieved so only the Getter is defined
  attr_reader :id 
```